aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_bedrock_patches.py

```diff
@@ -0,0 +1,202 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+# Modifications Copyright The OpenTelemetry Authors. Licensed under the Apache License 2.0 License.
+import abc
+import inspect
+from typing import Dict, Optional
+
+from opentelemetry.instrumentation.botocore.extensions.types import (
+    _AttributeMapT,
+    _AwsSdkCallContext,
+    _AwsSdkExtension,
+    _BotoResultT,
+)
+from opentelemetry.trace.span import Span
+
+
+class _BedrockAgentOperation(abc.ABC):
```

Comment:
Add a comment explaining why we need this:
* why we need a series of subclasses and a list of operation names
* why some operations are not covered by any subclass
* why some operations have multiple resources but we only map to one resource (e.g. AssociateAgentKnowledgeBase) and what rules we followed when making that choice.

In the comment, reference the API doc: https://docs.aws.amazon.com/bedrock/latest/APIReference/API_Operations_Agents_for_Amazon_Bedrock.html

Reply:
Added